src/test/java/com/multithreading/ThreadPoolTest.java

```diff
@@ -0,0 +1,94 @@
+package com.multithreading;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import java.util.concurrent.atomic.AtomicInteger;
+
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+public class ThreadPoolTest {
+    private ThreadPool threadPool;
+    private static final int NUM_THREADS = 4;
+
+    @BeforeEach
+    void setUp() {
+        threadPool = new ThreadPool(NUM_THREADS);
+    }
+
+    @AfterEach
+    void tearDown() {
+        threadPool.shutdown();
+    }
+
+    @Test
+    void testSubmitSingleTask() throws InterruptedException {
+        AtomicInteger counter = new AtomicInteger(0);
+        threadPool.submit(counter::incrementAndGet);
+
+        // Wait for a bit to allow the task to complete
+        Thread.sleep(100);
+
+        assertEquals(1, counter.get());
+    }
+
+    @Test
+    void testSubmitMultipleTasks() throws InterruptedException {
+        AtomicInteger counter = new AtomicInteger(0);
+        int numTasks = 10;
+
+        for (int i = 0; i < numTasks; i++) {
+            threadPool.submit(counter::incrementAndGet);
+        }
+
+        // Wait for a bit to allow all tasks to complete
+        Thread.sleep(500);
```

Comment:
i know what you are trying to do but FYI Google CL reviews would hammer that CL hard for having sleeps in tests :p If you couldn't sleep and you wanted to check that the job is completed how would you go about it?